Example App: add Slice Pizza DD App

tdimnet opened the pull request:
<!--  🎉 Hello there 🎉! Thank you for being a part of Datadog Apps! -->

## Motivation

-   What is the motivation / context for this change? (Text, screenshot, JIRA ticket, Github Issue)

<!-- - Is this a bugfix or a feature? -->

## Changes

-   What does this change? (Link to Github Issue if applicable, e.g. #44)

<!-- - If there's a UI Change: please include a screenshot. -->
<!-- - If there's not a UI Change: please remove this table. -->

| Before                         | After                        |
| ------------------------------ | ---------------------------- |
| ![Before alt text](Before URL) | ![After alt text](After URL) |

## Testing

<!--  Anything that would help a reviewer (or your future self) know if the change works as expected -->

## Releases

<!-- If you want to make a release at some point in the future, you'll need to add a changeset. -->
<!-- For more information, see: https://github.com/DataDog/apps/blob/master/RELEASE.md#package-releases. -->

Choose one:

-   [ ] No release is necessary.
        If you're only updating examples/documentation, this is likely what you want.
-   [ ] All packages that need a release have a changeset.

<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @datadog/create-app@2.0.1-canary.134.15e3345.0
  npm install @datadog/framepost@0.3.2-canary.134.15e3345.0
  npm install @datadog/ui-extensions-react@0.31.1-canary.134.15e3345.0
  npm install @datadog/ui-extensions-sdk@0.31.1-canary.134.15e3345.0
  # or 
  yarn add @datadog/create-app@2.0.1-canary.134.15e3345.0
  yarn add @datadog/framepost@0.3.2-canary.134.15e3345.0
  yarn add @datadog/ui-extensions-react@0.31.1-canary.134.15e3345.0
  yarn add @datadog/ui-extensions-sdk@0.31.1-canary.134.15e3345.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

Commit:
Setting up the project:

- Create the datadog app with widget and modals
- Integrate the API
- Create a proxy to handle CORS requests betweem the API and the client

Commit:
Setting up the project:

- Create the datadog app with widget and modals
- Integrate the API
- Create a proxy to handle CORS requests betweem the API and the client

Commit:
Building the form to add and remove pizza

Commit:
Building the form to add and remove pizza

Commit:
Add all API routes

Commit:
Add all API routes

Commit:
Get the order summary

Commit:
Get the order summary

Commit:
Adding cURL commands in README + updating proxy

Commit:
Adding cURL commands in README + updating proxy

Commit:
Updating all proxy routes

Commit:
Updating all proxy routes

Commit:
Proxy works fine

Commit:
Proxy works fine

Commit:
From the dd app, users can now:

- sign up
- sign in
- get token
- retrieve pizza lists with token

Commit:
From the dd app, users can now:

- sign up
- sign in
- get token
- retrieve pizza lists with token

Commit:
App completed, start cleaning code

Commit:
App completed, start cleaning code

Commit:
Clean code

Commit:
Clean code

Commit:
Run eslint and prettier

Commit:
Run eslint and prettier

Commit:
Fix warning issues and start adding css

Commit:
Fix warning issues and start adding css

Commit:
Continue adding style, images, and fixing litting issues.

Commit:
Continue adding style, images, and fixing litting issues.

Commit:
Add style

Commit:
Add style

Commit:
Add setup scripts

Commit:
Add setup scripts

Commit:
Run linter

Commit:
Run linter

Commit:
Run prettier and finish adding css

Commit:
Run prettier and finish adding css

ericrallen commented on examples/starter-kit/package.json:
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "datadog-app-example-starter-kit",
+    "name": "datadog-app-example-slice-pizza-app",
```
I think this one might have been overlooked.

ericrallen commented on examples/slice-pizza/datadog-app/app_manifest.json.bkp:
```diff
@@ -0,0 +1,98 @@
+{
```
Can we delete this file?

Commit:
Fix linter issues

Commit:
Fix linter issues

Commit:
Remove useless file and update name in package.json

Commit:
Remove useless file and update name in package.json

ericrallen commented on examples/slice-pizza/datadog-app/src/controller/index.ts:
```diff
@@ -0,0 +1,17 @@
+    <div>
+      The application controller is running in the background.
+    </div>
+    <a href='http://localhost:3000/widget'> Click here to open your widget </a>
```
Should this be `http://localhost:3002/slice-pizza-widget` instead?

Commit:
Clean code and write documentation

Commit:
Clean code and write documentation

Commit:
Update hero image (wrong folder)

Commit:
Update hero image (wrong folder)

Commit:
Adding hero

Commit:
Adding hero

Commit:
feat: add new UI to pizza example app

This updates the UI to use Chakra and a basic theme to easily adjust how we want to skin the pizza app.

It also adds a bootstrap.sh quality of life script to help new developers onboard into the example.

ericrallen commented on examples/slice-pizza/proxy/app.py:
```diff
@@ -0,0 +1,135 @@
+    return response.text
+
+
+@app.route('/api/cart', methods=['POST'])
```
I think that because the underlying `slice-life-pizzeria` API is doing asynchronous file manipulation when we send the `/api/cart` `POST` request, we may need to wait for it to actually finish here or we can encounter a race condition when adding many items to the cart.

Our UI will get resolved `Promises` from the Proxy, but the actual API might still be updating the storage file.

I tried to just convert this to `async def` and then `await` the `response`, but got an error about async support in Flask and couldn't quite figure out how to resolve it easily.

Commit:
chore: remove app-specific yarn.lock file

Commit:
feat: add header image to pizza widget modals

Commit:
fix: revert unrelated README change

Commit:
Merge branch 'ericrallen/slice-pizza-app-ui' into tdimnet/slice-pizza-app

Commit:
Run lint and prettier.

Commit:
Merge branch 'master' into tdimnet/slice-pizza-app

Commit:
Fix tests

Commit:
chore: update screenshot to new UI

ericrallen reviewed (approved):
This looks good to me.

I just pushed an update to the `hero.png` with the updated UI.

ericrallen commented:
**Preview**:

<img width="1512" alt="hero" src="https://user-images.githubusercontent.com/1667415/162817607-8e548cb6-2e66-40c0-930e-129ef4cb46dd.png">

Commit:
Add the Best Pizza venue in Denver info

Commit:
Merge branch 'tdimnet/slice-pizza-app' of github.com:DataDog/apps into tdimnet/slice-pizza-app

Commit:
Merge branch 'master' into tdimnet/slice-pizza-app

Commit:
Run prettier
